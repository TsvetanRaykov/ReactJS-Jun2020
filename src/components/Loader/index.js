import React from 'react'
import { Paper, withStyles, CircularProgress } from '@material-ui/core'

const styles = (theme) => ({
	main: {
		width: 'auto',
		display: 'block',
		marginLeft: theme.spacing(3),
		marginRight: theme.spacing(3),
		[theme.breakpoints.up(400 + theme.spacing(3 * 2))]: {
			width: 400,
			marginLeft: 'auto',
			marginRight: 'auto',
		},
	},
	paper: {
		marginTop: theme.spacing(8),
		display: 'flex',
		flexDirection: 'column',
		alignItems: 'center',
		justifyContent: 'center',
		height: 200,
	},
	loader: {
		color: theme.palette.primary.light,
	},
})

const Loader = (props) => {
	const { classes } = props
	return (
		<main className={classes.main}>
			<Paper className={classes.paper}>
				<CircularProgress className={classes.loader} disableShrink size={100} />
			</Paper>
		</main>
	)
}

export default withStyles(styles)(Loader)
