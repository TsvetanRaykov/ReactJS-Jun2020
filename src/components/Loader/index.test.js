import React from 'react'
import renderer from 'react-test-renderer'
import TestWrapper from '../../utils/test-wrapper'
import Loader from '.'

jest.mock('react', () => ({
	...jest.requireActual('react'),
	useLayoutEffect: jest.requireActual('react').useEffect,
}))

describe('<Loader />', () => {
	it('Should match the snapshot', () => {
		const wrapper = renderer
			.create(
				<TestWrapper>
					<Loader />
				</TestWrapper>
			)
			.toJSON()
		expect(wrapper).toMatchSnapshot()
	})
})
