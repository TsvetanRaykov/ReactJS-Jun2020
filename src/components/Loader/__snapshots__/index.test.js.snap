// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> Should match the snapshot 1`] = `
<main
  className="Loader-main-1"
>
  <div
    className="MuiPaper-root Loader-paper-2 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="MuiCircularProgress-root Loader-loader-3 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleDisableShrink MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</main>
`;
